src/Symfony/Component/PasswordHasher/Tests/Hasher/NativePasswordHasherTest.php

```diff
@@ -103,12 +103,13 @@ public function testBcryptWithNulByte()
         $hasher = new NativePasswordHasher(null, null, 4, \PASSWORD_BCRYPT);
         $plainPassword = "a\0b";
 
-        if (\PHP_VERSION_ID < 80218 || \PHP_VERSION_ID >= 80300 && \PHP_VERSION_ID < 80305) {
-            // password_hash() does not accept passwords containing NUL bytes since PHP 8.2.18 and 8.3.5
-            $this->assertFalse($hasher->verify(password_hash($plainPassword, \PASSWORD_BCRYPT, ['cost' => 4]), $plainPassword));
-        }
+        try {
+            $verified = $hasher->verify($hasher->hash($plainPassword), $plainPassword);
 
-        $this->assertTrue($hasher->verify($hasher->hash($plainPassword), $plainPassword));
+            $this->assertTrue($verified);
+        } catch (\ValueError $error) {
```

Comment:
Please keep the `try` block minimal and only include the call that you expect to raise the `ValueError`. Given that `ValueError` is a very generic exception class, we should also sniff the exception message to make sure we're skipping the test for the right reasons.

Please also add a clarifying comment hinting at the PHP commit that caused this change.

I think, we should also skip the test for PHP >= 8.4 entirely (via annotation). This way, we can find and delete it once we open a branch that doesn't support PHP 8.3 anymore (likely Symfony 8.0).